docs/LSPSupport.md

```diff
@@ -389,29 +389,31 @@ showing automatic folding of both the file header comment and imports when the f
 
 ![textDocument/foldingRange_collapseByDefault](./images/lsp-support/textDocument_foldingRange_collapseByDefault.gif)
 
-#### Code block provider
+### Selection range
+
+[textDocument/selectionRange](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_selectionRange) is implemented with
+the `extendWordSelectionHandler` extension point and used via the IDE's **Extend Selection** (**Ctrl+W** on Windows/Linux; **Opt+Up** on Mac) and **Shrink Selection** (**Ctrl+Shift+W** on Windows/Linux; **Opt+Down** on Mac) actions.
+
+Here is an example with the [TypeScript Language Server](https://github.com/typescript-language-server/typescript-language-server) showing **Extend/Shrink Selection** using `textDocument/selectionRange`:
+
+![textDocument/selectionRange](./images/lsp-support/textDocument_selectionRange.gif)
 
-Additionally LSP4IJ registers the an implementation of the `codeBlockProvider` extension point with
+### Code block provider

```

Comment:
I also updated the code block provider documentation to clearly state that code blocks are derived using selection ranges first and then folding ranges if necessary.